examples/terraform-aws-lambda-example/main.tf

```diff
@@ -0,0 +1,39 @@
+# ---------------------------------------------------------------------------------------------------------------------
+# AWS LAMBDA TERRAFORM EXAMPLE
+# See test/terraform_aws_lambda_example_test.go for how to write automated tests for this code.
+# ---------------------------------------------------------------------------------------------------------------------
+
+provider "archive" {
+  version = "1.3"
+}
+
+data "archive_file" "zip" {
+  type        = "zip"
+  source_dir  = "${path.module}/src"
+  output_path = "/tmp/.${var.function_name}.zip"
```

Comment:
NIT: This won't work on all OS platforms (e.g Windows). Probably better to use `${path.module}`.